paltas/Analysis/dataset_generation.py

```diff
@@ -511,3 +511,50 @@ def rotation_generator(dataset):
 
 	# Return a rotation generator on our base dataset.
 	return rotation_generator(base_dataset)
+
+
+def generate_params_as_input_dataset(base_dataset,params_as_inputs,
+	all_params):
+	"""Generate a dataset where some of the lens parameters are treated as
+	inputs to the model.
+
+	Args:
+		base_dataset (generator): A generator that yields a tuple containing
+			the images and the parameters values.
+		params_as_inputs ([str,...]): A list of strings containing the
+			parameters that will be turned into inputs for the model.
+		all_params ([str,...]): A list of strings containing the parameters
+			that will be outputted by the generator.
+
+	Returns:
+		(generator): A generator that returns a tuple with the inputs (which
+		are the image and the scalar input values), and the outputs parameter
+		values.
+	"""
+
+	# Which inputs do we need to pull out and which ones do we keep as
+	# learning parameters.
+	pai_indices = [all_params.index(param) for param in params_as_inputs]
```

Comment:
This had me confused for a bit. So pai_indices are the parameter as input, and pai_mask is a boolean array that selects all the other params? If I got that right, maybe clearer to do e.g.
```
param_is_input = np.array([param in params_as_inputs for param in all_params])
param_is_output = ~param_is_input
```
Currently, it's a bit odd that one set gets selected through their indices and the other through a boolean array.

Reply:
You're right, this is much clearer.